### PR TITLE
-update outdated api reference section for messaging and optimize

### PR DESCRIPTION
**Pull request by sbenedicadb**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/pages/resources/migration/ios/migrate-to-5x.md [hunk 1]
```diff
@@ -71,3 +71,11 @@ Once the previous command is complete, run `pod install` or `pod update` to upda
 If you are using Swift Package Manger (SPM) for managing your app dependencies, you can now include the Experience Platform 5.x SDKs either through Xcode UI, or by declaring them as dependencies in the Package.swift project file. For more details, follow the guide for [managing dependencies using Swift Package Manager](../../manage-spm-dependencies.md).
 
 ## Update outdated API references
+
+#### Messaging
+
+Change usages of `Messaging.handleNotificationResponse(_:applicationOpened:withCustomActionId:)` and `Messaging.handleNotificationResponse(_:closure:)` to `Messaging.handleNotificationResponse(_:urlHandler:closure:)`
```
Comment (praveek) [inline src/pages/resources/migration/ios/migrate-to-5x.md:77]: Is the API reference in Messaging extension updated with this new API? If so, can you link to that API?

File: src/pages/resources/migration/ios/migrate-to-5x.md [hunk 2]
```diff
@@ -71,3 +71,11 @@ Once the previous command is complete, run `pod install` or `pod update` to upda
...
+
+#### Optimize
+
+Change all references from class `Proposition` to `OptimizeProposition`.
```
Comment (praveek) [inline src/pages/resources/migration/ios/migrate-to-5x.md:81]: Same as above. The API should also be updated in https://developer.adobe.com/client-sdks/edge/adobe-journey-optimizer-decisioning/api-reference/#public-classes

File: src/pages/resources/migration/ios/migrate-to-5x.md (hunk 1 above)
Comment (emdobrin) [reply src/pages/resources/migration/ios/migrate-to-5x.md:77]: @sbenedicadb can we have some comparison and/or examples on how customers should implement these APIs for migration?

File: src/pages/edge/adobe-journey-optimizer/release-notes.md [hunk 3]
```diff
@@ -11,16 +11,16 @@ keywords:
...
+* Added [`handleNotificationResponse(_:urlHandler:closure:)`](./../../edge/adobe-journey-optimizer/api-reference/#handlenotificationresponse) API to provide more control over URL handling from `UNNotificationResponse` objects.
   * Removed deprecated public API `Messaging.handleNotificationResponse(_:applicationOpened:withCustomActionId:)`.
   * Removed public API `Messaging.handleNotificationResponse(_:closure:)` API.
 * Added support for [Code Based Experiences](https://experienceleague.adobe.com/en/docs/journey-optimizer/using/code-based-experience/get-started-code-based).
-  * Added public API `Messaging.updatePropositionsForSurfaces(_)`.
-  * Added public API `Messaging.getPropositionsForSurfaces(_:completion:)`.
+  * Added public API [`Messaging.updatePropositionsForSurfaces(_)`](./../../edge/adobe-journey-optimizer/code-based/api-reference/#updatepropositionsforsurfaces).
+  * Added public API [`Messaging.getPropositionsForSurfaces(_:completion:)`](./../../edge/adobe-journey-optimizer/code-based/api-reference/#getpropositionsforsurfaces).
```
Comment (praveek) [inline src/pages/edge/adobe-journey-optimizer/release-notes.md:23]: nit: All these links can be simplified as they are within the same directory.

File: src/pages/edge/adobe-journey-optimizer/tabs/api-reference.md [hunk 4]
```diff
@@ -74,25 +74,25 @@ public static void handleNotificationResponse(final Intent intent,
 
 ```swift
 static func handleNotificationResponse(_ response: UNNotificationResponse,
```
Comment (praveek) [inline src/pages/edge/adobe-journey-optimizer/tabs/api-reference.md:76]: Can we still keep the old API with deprecated / removed warning? We decided to do it for other APIs like registerExtension which will be removed with Android 3.0.0.

**Review by calebk1 — APPROVED** [review]
Content-wise LGTM!

**Pull request merged**